trussworks/react-uswds
# fix: a11y: search component submit accessible naming

# Summary

adds support for accessible name for search submit button in small search variant, and enhances naming capabilities on all search variants submit button to prevent duplicate naming issues on pages with multiple search components. Example of this scenario in USSF portal provided in screen shots.

## Related Issues or PRs

- [Space Force Portal repo small search accessibility branch](https://github.com/USSF-ORBIT/ussf-portal-client/pull/1205), needs accessible name for button, and requires it be unique so the button name does not clash with site search field in header.

## How To Test


check output in ANDI, which should reflect the fallback name of "Search". Note, the aria label added to the non-small variants will not conflict with the typed name button text in the default instances, because the aria label will override the button text in the accessibility tree. This will allow us to have "search" appear visually for users who will have visual context, but more precise labeling for users of assistive tech.

I encourage reviewers to be vigilant in the changes made to the Search test file. My intention is to pull this into the USSF portal repo to ensure it works as expected.

### Screenshots (optional)

#### Example of multiple searches in a single interface

<img width="1197" alt="image" src="https://github.com/trussworks/react-uswds/assets/59394696/5167375b-0f0e-4951-9b2f-ce7fd094b92f">

#### ANDI outputs in local storybook for react-uswds

##### Buttons with inner text

<img width="529" alt="image" src="https://github.com/trussworks/react-uswds/assets/59394696/b94670ee-582b-488b-a9fd-ee42bb4b1b17">


##### Small Search: Buttons without Inner Text

<img width="522" alt="image" src="https://github.com/trussworks/react-uswds/assets/59394696/ad6e0edb-9ef0-4e1d-b50d-0f3d0aa51c76">
